.evergreen/evergreen.yml.in

```diff
@@ -1137,22 +1165,26 @@ buildvariants:
     tasks:
       - name: compile_artifact
 
-  - name: pkg_smoke_tests_docker
-    display_name: "package smoke tests (Docker)"
+  - name: pkg_smoke_tests_docker_x64
+    display_name: "package smoke tests (x64 Docker)"
     run_on: ubuntu1804-small
     tasks:
-      - name: pkg_test_docker_centos7_rpm
-      - name: pkg_test_docker_rocky8_rpm
-      - name: pkg_test_docker_fedora34_rpm
-      - name: pkg_test_docker_suse12_rpm
-      - name: pkg_test_docker_suse15_rpm
-      - name: pkg_test_docker_amazonlinux1_rpm
-      - name: pkg_test_docker_amazonlinux2_rpm
-      - name: pkg_test_docker_debian9_deb
-      - name: pkg_test_docker_debian10_deb
-      - name: pkg_test_docker_debian11_deb
-      - name: pkg_test_docker_ubuntu18_04_deb
-      - name: pkg_test_docker_ubuntu20_04_deb
+  <% for (const { distributionBuildVariants, executableOsId } of EXECUTABLE_PKG_INFO) {
```

Comment:
I was wondering, is there a way to move this heavy logic outside of yaml? Seems hard to understand the code here and also a bit intimidating to change.

If we are using something like lodash.template or similar I'd prefer something like this with more helpers:

```
<% for ( const dockerfile of getSmokeTestDockerfiles(EXECUTABLE_PKG_INFO)) { %>
  - name: pkg_test_<%= asTaskName('pkg_test', dockerfile) %>
<% } %>
```

Reply:
Mh yeah, we could look into using `lodash.template` at some point, and we can also add helper functions for simplifying this.

It’s definitely easier not to outsource everything from the YAML here though; it has been very helpful to have a single source of truth for how our CI jobs are defined.

Reply:
Yeah, that's a good point, I mainly find it hard to parse, probably just having helpers to make the code "lighter" would be enough.

Reply:
It absolutely is hard to parse! I don’t want to act like it isn’t :) 

I’ll definitely think about it, for now my main goal here is getting everything to work tbh